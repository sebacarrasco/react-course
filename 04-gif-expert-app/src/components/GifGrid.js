import React from 'react'
import {GifGridItem} from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ( { category } ) => {

    // const [images, setImages] = useState([]);

    // // // Cuando cambie un elemento del arreglo que se le pasa
    // // // al useEffect, entonces se volver치 a aejecutar getGifs
    // // // (En este caso solo se ejecutar치 solo una vez ya que
    // // // category no cambia nunca)
    // useEffect( () => {
    //     getGifs(category)
    //         .then(setImages);
    // }, [category]);

    // Lo que est치 comentado arriba es equivalente a lo siguiente
    // (usando un custom hook que por detr치s usa useEffect)
    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p>Loading...</p>}
            <div className="card-grid">
                    {
                    images.map(img => 
                            <GifGridItem key={img.id} {...img}/> // Estamos haciendo spread del objeto img para
                                                                // para pasar sus propiedades "sueltas"
                    )
                    }
            </div>
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};