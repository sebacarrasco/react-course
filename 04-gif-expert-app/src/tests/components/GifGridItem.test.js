import React from 'react';
import {GifGridItem} from "../../components/GifGridItem";
import {shallow} from "enzyme";

describe('Tests for <GifGridItem />', () => {
    
    const title = "A title for the GifGridItem";
    const url = "https://example_url.com";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should render the component correctly', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('should have a parragraph with the title', () => {
        const parragraphText = wrapper.find("p").text();
        
        expect(parragraphText.trim()).toBe(title);
    });
    
    test('should hace the image with src=url and alt=title', () => {
        const image = wrapper.find("img");
        const {src, alt} = image.props();

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should have animate__bounce css class', () => {
        const div = wrapper.find("div");
        const {className} = div.props();

        expect(className).toContain("animate__bounce");
        // An√°logo a hacer: expect(className.includes("animate__fadeIn")).toBe(true);
    });
    
    

});