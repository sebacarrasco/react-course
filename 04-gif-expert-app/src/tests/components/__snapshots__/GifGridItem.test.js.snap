// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGridItem /> should render the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="A title for the GifGridItem"
    src="https://example_url.com"
  />
  <p>
     
    A title for the GifGridItem
     
  </p>
</div>
`;
